extensions_admin/pulp_rpm/extensions/admin/upload/comps.py

```diff
@@ -0,0 +1,40 @@
+from gettext import gettext as _
+
+from pulp.client.commands.repo.upload import UploadCommand
+from pulp_rpm.common.ids import TYPE_ID_PKG_GROUP
+
+
+NAME_XML = 'comps'
+DESC_XML = _('uploads comps.xml into a repository')
+SUFFIX_XML = '.xml'
+
+
+class _CreateCompsCommand(UploadCommand):
+    """
+    Base command for uploading comps.xml. This shouldn't be instantiated directly
+    outside of this module in favor of one of the type-specific subclasses.
+    """
+
+    def __init__(self, context, upload_manager, type_id, suffix, name, description):
+        super(_CreateCompsCommand, self).__init__(context, upload_manager, name=name,
+                                                  description=description)
+        self.type_id = type_id
+        self.suffix = suffix
+
+    def determine_type_id(self, filename, **kwargs):
+        return self.type_id
+
+    def matching_files_in_dir(self, directory):
+        all_files_in_dir = super(_CreateCompsCommand, self).matching_files_in_dir(directory)
+        comps = [f for f in all_files_in_dir if f.endswith(self.suffix)]
```

Comment:
in case there are such, it will just try to parse the xml file and will not find any units( i.e groups, categories, environments) to save.